File: !ЛАБА 4!/PayrollAppWx/Employee.cpp
#include "Employee.h"

Employee::Employee(int empId, const std::string& first, const std::string& last, double salary)
    : id(empId), firstName(first), lastName(last), baseSalary(salary)
{
}

int Employee::GetId() const { return id; }
std::string Employee::GetFirstName() const { return firstName; }
std::string Employee::GetLastName() const { return lastName; }
double Employee::GetSalary() const { return baseSalary; }

void Employee::SetName(const std::string& first, const std::string& last)
{
    firstName = first;
    lastName = last;
}

void Employee::SetSalary(double salary)
{
    baseSalary = salary;
}

double Employee::CalculateSalary() const
{
    return baseSalary;
}

File: laba4/Employee.cs
using System;
using System.Collections.Generic;

namespace PayrollApp
{
    public class Employee
    {
        private int id;
        private string firstName;
        private string lastName;
        private double baseSalary;
        private List<CompletedWork> completedWorks;

        public Employee(int empId, string first, string last, double salary = 0)
        {
            id = empId;
            firstName = first;
            lastName = last;
            baseSalary = salary;
            completedWorks = new List<CompletedWork>();
        }

        public void AddCompletedWork(CompletedWork work)
        {
            completedWorks.Add(work);
        }

        public double CalculateSalary()
        {
            return baseSalary;
        }

        public int GetId() => id;
        public string GetFirstName() => firstName;
        public string GetLastName() => lastName;
        public List<CompletedWork> GetCompletedWorks() => completedWorks;

        public void SetName(string first, string last)
        {
            firstName = first;
            lastName = last;
        }

        public void SetSalary(double salary)
        {
            baseSalary = salary;
        }

        public double GetSalary() => baseSalary;
    }
}